blog/12-07-2022-solidity-from-zero-to-hero/solidity-from-zero-to-hero.md

```diff
@@ -0,0 +1,468 @@
+
+```solidity
+//SPDX-license-Identifier: GPL-3.0
+
+pragma solidity^0.4.13
+
+contract MyContract {
+	// everything else inside this block 😁😁
+}
+```
+
+## Functions
+
+Functions are the basic building blocks of contracts. A function has the following form:
+
+```solidity
+function name(argument_1, argument_2, ...) { 
+	// Body of the function 
+}
+```
+
+```solidity
+function multiply(uint a, uint b) returns (uint c) {
+	c = a * b;return c;
+}
+```
+
+# **Data types**:
+
+There are six data types: uint, int, bool, string, address, and bytes32.
+
+## Unsigned Integer (uint)
+
+The unsigned integer (uint) is used to store unsigned integers (positive numbers) in solidity. The memory allocation of this data type can be tweaked by changing the number after it (in order of 8, 16, 24, …, 256) e.g., uint256 can store numbers from 0 - 2 ^ 256 - 1 while uint8 ranges from 0 to 2^8 -1. The type uint is an alias of uint256; It has a range that makes it suitable for representing values like token balances, exchange rates, and other numerical data.
+
+```solidity
+contract MyContract {
+	uint256 public someNumber;
+	
+	function change() public {
+		someNumber = 5;
+	}
+}
+```
+
+### Signed Integer (int)
+
+The signed integer (int) can be used to store signed integers (negative and positive numbers) in solidity. Just like the uint, memory allocation of this data type can be tweaked by changing the number after it (in order of 8, 16, 24, …, 256) e.g., uint256 can store numbers from 2 ^ 255 - 2 ^ 255. The type int is an alias of int256.
+
+```solidity
+contract MyContract {
+	int256 public someNumber;
+	
+	function change() {
+		someNumber = -5;
+	}
+}
+```
+
+### Boolean (bool)
+
+The boolean data type can be used to store either a true or false value. Booleans can be used in conditional statements, such as if statements. For example, the following code will only execute the second statement if the boolean is true:
+
+```jsx
+if (bool) { 
+	// Statements to execute if bool is true.
+} else { 
+	// Statements to execute if bool is false.
+} 
+```
+
+Another example:
+
+```solidity
+contract Bool {
+	bool public value;
+
+	function setValue(bool newValue) {
+		value = newValue;
+	}
+
+	function getValue() constant returns (bool) {
+		return value;
+	}
+}
+```
+
+### String
+
+A string is simply a series of characters enclosed in quotation marks.
+
+```solidity
+string sample = "this is a sample string";
+```
+
+### Address
+
+An address is a 20-byte hexadecimal number that represents a participant on the Ethereum blockchain. The Address is generated by Remix like this:
+
+```solidity
+address myAddress = 0xA870daB9b7c72C3d9a7b728Ae6Bf1f359D5F2773; //
+```
+
+## Modifiers
+Modifiers are a powerful tool in Solidity that allow you to control who can call a function and when they can call it. You can use a modifier, for example, to ensure that a function can only be called by the contract owner.
+
+```solidity
+pragma solidity ^0.5.0;
```

Comment:
Please format the code properly here.

- Remove extra lines in between the code.
- Instead of tabs for code indentation use four spaces for consistency.